Add Privacy to docs

imestin opened the pull request:
Closes Dedicated Privacy page #816

imestin commented:
This is my base idea of this. I will add "read more" links and extend the text a little bit, but this would be the general structure. Do you think this is good direction?

xaur commented:
I really like the general direction of thinking here, the intent to bring crypto privacy considerations into people's awareness, and the contrast with a bank account. The (biggest?) problem with privacy is people are not aware of what's going on with their data. Using simple language is a bridge that helps them get started. In crypto it is more important than earlier because money is on the line. Ultimately, this is about sovereignty too, although "sovereignty" is yet another thing that people don't often think about.

> newly generated addresses even on a website

Not clear what this refers to.

> For using a VSP, not using the same address is not possible yet.

Maybe even with solo? This needs checking.

> If we only spend fraction of an UTXO

This made me think about the target audience of this doc, or in other words, knowledge level required to read this. This bit implies the reader is familiar with what is an "output" and what is an "UTXO".

I'm not immediately suggesting to define all terms from scratch in this doc, rather to think one step higher and consider the "requirements" for this doc.

Re identification, I would add that many exchanges today voluntarily feed chain analysis companies with their customer data. This creates an interesting snowball dynamic where such company's database is enriched with every client (that must feed the data by agreement). Such databases only grow and become more and more powerful. I bet nothing is ever deleted from them.

> [General-Security](https://docs.decred.org/advanced/general-security/)

I guess you don't need a full absolute URL for local links, but dcrdocs people should know better.

imestin commented:
Thank you for your feedback.



>     newly generated addresses even on a website
I meant, if you have a website where people can send you money, probably you should use a script that generates new addresses every time. I will re-frame this somehow.

I will do a little bit more research about VSPs.

About UTXOs, yes, I was not really sure whether this needs explanation or not. I'm not sure indeed whether this doc should be under "Advanced". What does "Advanced" mean in the docs? What level of knowledge we expect?

s-ben commented:
@imestin, the Advanced section is something of a dumping ground for pages that don't fit elsewhere. Later, we may reorg the docs and come up with a better place for this, but Advanced is probably the best place for this page now. For a point of comparison, this feels similar to the [Security](https://docs.decred.org/advanced/general-security/) page.

imestin commented:
I changed a little bit the `newly generated addresses even on a website` part and added a link to an outside resource that explains UTXOs. I don't want to explain UTXOs inside the docs, but if you think it's necessary I can.

Soon I will add more updates.

imestin commented:
I think now it covers the things a basic user should know about.
I don't know if the relative links will work when it's uploaded to the website.

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+# General information on Privacy
+
+## About transaction graphs
```
Capitalise Transaction Graphs

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+## About transaction graphs
+
+It’s important to understand, that transactions are pseudonymous, but not anonymous. That means, a transaction graph can be built up, and there exists companies that are doing that work. If at some point one of our transactions is connected to our real world identity, it becomes easy to make assumptions based on that, for example that other transactions also belong to us, or that a specific unspe ...
+By  checking the status of our transaction on a blockchain explorer, our IP address can be connected to one of our wallet addresses, this could create a starting point for a transaction graph. A possible solution for this is to use blockchain explorer only over Tor.
```
Added some extra clarification for this paragraph:

> One possible way your personal identity could be linked to your blockchain activity is through the use of a block explorer. The administrator of the block explorer may be able to view your IP address and which addresses or transactions you are browsing, and they may infer that you own those addresses or transactions. A possible solution for this is to use blockchain explorers exclusively over Tor.

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+
+## Transaction broadcasting
+
+Broadcasting the transaction can reveal that the transaction you are broadcasting belongs to you. A solution for this is to broadcast your transaction through a website, over Tor. For example, [https://explorer.dcrdata.org/decodetx](https://explorer.dcrdata.org/decodetx)
```
Broadcasting **a** transaction* 


also, rather than duplicating the URL using \[]\(), we can just use <>.

`<https://explorer.dcrdata.org/decodetx>`

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+
+## Trade-offs of reusing vs not reusing addresses
+
+For privacy, the best practice is to not reuse an address ever. A cryptocurrency address can be thought of as a cryptocurrency invoice instead. The term "address" does not mean, that it was designed to reuse it, like one would do with a bank account number for example. If someone would really care about privacy he or she would use a new receiving address even on a website, generated by a script. 
```
The comma after "does not mean" is unnecessary

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+
+For privacy, the best practice is to not reuse an address ever. A cryptocurrency address can be thought of as a cryptocurrency invoice instead. The term "address" does not mean, that it was designed to reuse it, like one would do with a bank account number for example. If someone would really care about privacy he or she would use a new receiving address even on a website, generated by a script. 
+On the other hand, address reuse is preferred in situations when privacy is not important and we don't want to make the mistake of sending over large amount of funds to a wrong address. Reusing addresses is also preferred when transparency is important.
+For POS voting, not using the same address is not possible yet. 
```
I would remove this line about PoS voting because its not strictly true. Also it will become redundant when privacy is implemented into VSPs

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+## Trade-offs of reusing vs not reusing addresses
+
+For privacy, the best practice is to not reuse an address ever. A cryptocurrency address can be thought of as a cryptocurrency invoice instead. The term "address" does not mean, that it was designed to reuse it, like one would do with a bank account number for example. If someone would really care about privacy he or she would use a new receiving address even on a website, generated by a script. 
+On the other hand, address reuse is preferred in situations when privacy is not important and we don't want to make the mistake of sending over large amount of funds to a wrong address. Reusing addresses is also preferred when transparency is important.
```
> On the other hand, there may be situations where address reuse may be preferred, or even necessary. The Decred Treasury is a good example of this. Transparency and auditability are more important than privacy in the context of the treasury.

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+
+## Change address
+
+If we only spend fraction of an [UTXO](https://coincentral.com/utxo-beginners-explainer/), a change output will be generated. This is not good for privacy. For example, the change UTXO will be merged with another unspent transaction output next time when we are spending money, thus creating a signal that both of the transactions belong to us. An average wallet software won’t give us information a ...
```
**a** fraction*

jholdstock replied:
Need to explain that UTXO == unspent transaction outputs. We will normally introduce an acronym like this:

> Unspent Transaction Output (UTXO)

Then we can freely use "UTXO" and the user will know what it means

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+
+Many times this is something physical, for example leaving behind a printed QR code somewhere, writing our nickname or real name on a piece of paper. Your name can be inprinted on the next page of the sticky notes stack, for example. Or that could also happen, that you leave behind your phone or computer unlocked for some minutes, with important information displayed on screen.
+
+## Centralized mixers
```
This section seems to be completely focused on bitcoin. Can we please either make this Decred-centric, or remove it.

imestin replied:
I included these paragraph to make the reader think about privacy tech. I was not trying to direct the reader to start using these tools, I included them for historic reasons mainly. I know that it is a bit focused on bitcoin, but I don't know how otherwise to make the reader think about the basic concept.

I can try to re-frame this paragraph, but the basic intention is not to give present-day information in this section. CoinShuffle++ was meant to be present day information.

imestin replied:
I shortened the Centralized Mixers section

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+
+## Non-custodial coin mixing
+
+It is possible to mix coins without a centralized exchange. One method is [CoinJoin](https://en.bitcoin.it/wiki/CoinJoin), which will combine multiple payments from multiple spenders into a single transaction.
```
We should link to Decred implementations of coinjoin/coinswap. Otherwise this is not relevant.

imestin replied:
The reason I included it is that I wanted the reader to think about what kind of things exist in this field. I didn't wanted him/her to start using these tools, just to start thinking about how privacy tech can work.

imestin replied:
https://cspp.decred.org/

I found this, but as I know, CoinShuffle++ and CoinJoin is not exactly the same. I wouldn't link to this resource.

What do you think?

imestin replied:
I removed CoinJoin and CoinSwap

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+
+## Using the Lightning Network for privacy
+
+Privacy attacks like [common-input-ownership heuristic](https://en.bitcoin.it/wiki/Privacy#Common-input-ownership_heuristic), [address reuse](https://en.bitcoin.it/wiki/Privacy#Address_reuse) and change address detection fundamentally don’t work on any off-chain transactions technology. The Lightning Network is an off-chain transaction technology. One possible downside is that the operator of the ...
```
As a reader, I am wondering what information can a lightning node log, and why is that a privacy risk?

jholdstock commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,63 @@
+
+## Platform-related questions
+
+Keep in mind that the platform you are using can also have vulnerabilities. A VPN does not provide any means of defense if the system itself that you are using is hacked. Try to use open-source software, and if possible, open-source hardware as well. See [General-Security](general-security) for more information.
```
What do you mean by platform here? Operating system?

Link to `general-security.md` with markdown file extension.

xaur commented:
I suggest to use web page links instead of PDF links for the paper. This way the reader can click and see a summary in his browser before downloading the PDF

https://arxiv.org/abs/1502.01657
https://eprint.iacr.org/2016/701

and so on

imestin commented:
@xaur but you can not actually read the research paper when you click on these links. For reading you will have to download the PDF, or you can open the PDF with your browser, if your browser supports this.

On the other hand, this way the user could see what he is about to download, before actually downloading.

I wouldn't change it.

imestin commented:
what do you think about the issue with the centralized mixers? Do you think it's good idea to provide information which is historical, or would you only provide Decred-specific info?

xaur commented:
> but you can not actually read the research paper when you click on these links. For reading you will have to download the PDF

It is trivial to find the PDF download on those pages, but it is not trivial to go in the opposite direction, i.e. to change the PDF link into the abstract link. I think it's useful to read the abstract before downloading the paper.

re centralized mixers, I think the Decred-focused page should only include information that is helpful to understand Decred's implementation. In that context, mentioning existing mixer tech (even for Bitcoin) is ok IMO. However, any significant exploration of non-Decred mixing tech should live in a separate page about general privacy in crypto.

dajohi commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,62 @@
+## Non-custodial Coin Mixing
+
+It is possible to mix coins without a centralized exchange. One method is [CoinJoin](https://en.bitcoin.it/wiki/CoinJoin), which will combine multiple payments from multiple spenders into a single transaction.
+Other method is [CoinSwap](https://en.bitcoin.it/wiki/CoinSwap), which is using atomic swaps. A third method is [TumbleBit](https://en.bitcoin.it/wiki/TumbleBit), which allows parties to make payments through an untrusted Tumbler.
```
could reference github.com/decred/tumblebit and github.com/decred/cspp around here...

imestin commented:
I will create privacy folder, as suggested in the post from @xaur 

> There are two ways to organize this.

First is group by aspect: create new Privacy group as suggested above and let it host both general privacy and Decred privacy. This is similar to existing Governance group.

The other way I didn't imagine initially is to fit into existing grouping by use case. Place the general privacy page under Advanced (next to Security), and then place another page for Decred privacy features in Wallets category, similar to how it hosts a page for Lightning Network use case.

xaur commented:
@imestin please not that I'm just lurking around while @jholdstock and @s-ben actively maintain this repo. I hope they follow this issue to prevent wasted work.

gozart1 commented on docs/advanced/general-privacy.md:
```diff
@@ -0,0 +1,61 @@
+
+## Avoid Leaving Identifiers in Public Places
+
+Many times this is something physical, for example leaving behind a printed QR code somewhere, writing our nickname or real name on a piece of paper. Your name can be inprinted on the next page of the sticky notes stack, for example. Or that could also happen, that you leave behind your phone or computer unlocked for some minutes, with important information displayed on screen.
```
I noticed the word "our" and "your" is used interchangeably throughout this file. For example it says "our nickname" here and that just seems off to me. Suggest just sticking to "your" in all cases.
